x-pack/solutions/security/plugins/security_solution/server/endpoint/routes/resolver/entity/utils/supported_schemas.test.ts

```diff
@@ -0,0 +1,393 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getSupportedSchemas } from './supported_schemas';
+import type { ExperimentalFeatures } from '../../../../../../common';
+import * as securityModules from './security_modules';
+
+jest.mock('./security_modules', () => ({
+  CORE_SECURITY_MODULES: [
+    'crowdstrike',
+    'jamf_protect',
+    'sentinel_one',
+    'sentinel_one_cloud_funnel',
+  ],
+  MICROSOFT_DEFENDER_MODULES: ['microsoft_defender_endpoint', 'm365_defender'],
+  getSecurityModuleDatasets: jest.fn(),
+  getAllSecurityModules: jest.fn(),
+}));
+
+const mockGetSecurityModuleDatasets =
+  securityModules.getSecurityModuleDatasets as jest.MockedFunction<
+    typeof securityModules.getSecurityModuleDatasets
+  >;
+
+describe('getSupportedSchemas', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    // Set up default mock implementation to return realistic data
+    mockGetSecurityModuleDatasets.mockImplementation((modules: readonly string[]) => {
+      const datasetMap: Record<string, string[]> = {
+        crowdstrike: ['crowdstrike.alert', 'crowdstrike.falcon', 'crowdstrike.fdr'],
+        jamf_protect: [
+          'jamf_protect.telemetry',
+          'jamf_protect.alerts',
+          'jamf_protect.web-threat-events',
+          'jamf_protect.web-traffic-events',
+        ],
+        sentinel_one: ['sentinel_one.alert'],
+        sentinel_one_cloud_funnel: ['sentinel_one_cloud_funnel.event'],
+        microsoft_defender_endpoint: ['microsoft_defender_endpoint.log'],
+        m365_defender: ['m365_defender.alert', 'm365_defender.incident'],
+      };
+
+      const datasets: string[] = [];
+      modules.forEach((module) => {
+        const moduleDatasets = datasetMap[module];
+        if (moduleDatasets) {
+          datasets.push(...moduleDatasets);
+        }
+      });
+      return datasets;
+    });
+  });
+  describe('Microsoft Defender for Endpoint integration', () => {
+    it('should include Microsoft Defender datasets when feature flag is enabled', () => {
+      const experimentalFeatures: ExperimentalFeatures = {
+        microsoftDefenderEndpointDataInAnalyzerEnabled: true,
+      } as ExperimentalFeatures;
+
+      const schemas = getSupportedSchemas(experimentalFeatures);
+      const filebeatSchema = schemas.find((schema) => schema.name === 'filebeat');
+
+      expect(filebeatSchema).toBeDefined();
+      expect(filebeatSchema!.constraints).toEqual(
```

Comment:
Isn't this assertion comparing mocked values to hardcoded ones? 
It feels like we have a lot of duplication between the mock implementation and the hardcoded expectations.